### PR TITLE
Adding Helm env vars where XDG exposed

### LINKED ISSUE #7919: Override path for helm data
The only way to specify the path to where data (same for cache and config) are suppose to live is either using the default path (that uses the $HOME variable) or by setting the XDG_DATA_HOME.
This is a problem if you not want to use the default path and can't use the XDG_DATA_HOME override (e.g. because of some dependency that uses the same variable)

Suggestion:
Add a second way of overriding the default path. A new variable (something like `HELM_DATA_HOME`) that overrides both the default path AND the XDG_DATA_HOME variable.

This would be useful in github workflow which overwrites the HOME variable and attaches a volume to that path when executing a container such that plugins installed in the home folder at build time will be lost.

### PR DESCRIPTION
**Pull request by mattfarina**
Helm had been exposing XDG based variables to end users. This lead
to confusion. For example, if a user wanted to change the cache
location Helm used should they change the XDG variable? Since this
would be like changing the HOME environment variable the answer
is no.

This change adds HELM_*_HOME environment variables to be used
in addition to XDG ones of the same name. Helm will now look
for the Helm specific variable. If not set, Helm will fall
back to XDG locations. If those are not set a default location
will be used. This keeps XDG in use as a default when present,
provides users with the ability to set the location, and removes
XDG from being exposed to end users to avoid confusion.

Closes #7919

@marckhouzam Would this solve for #6890, too?

### CODE REVIEW

**Review by marckhouzam — COMMENTED** [review]
I like this.  It's simple and gives the proper flexibility to users.
I think I'm too new to approve adding new env variables that will become permanent, so I'll abstain from giving official approval, but you do have my vote.

As for #6890, it resolves the second point, but the first point is not addressed, as can be better explained by https://github.com/helm/helm-www/issues/336 

Some small nits as comments.

Thanks for addressing this.

File: cmd/helm/root.go [hunk 1]
```diff
@@ -46,20 +46,20 @@ Environment variables:
...
 +------------------+--------------------------------------------------------------------------------------------------------+
 
-Helm stores configuration based on the XDG base directory specification, so
+Helm stores cache, configuration, and data based on configuration:
 
-- cached files are stored in $XDG_CACHE_HOME/helm
-- configuration is stored in $XDG_CONFIG_HOME/helm
-- data is stored in $XDG_DATA_HOME/helm
+- If a HELM_*_HOME environment variable is set that is used
+- Otherwise, on systems supporting the SDG base directory specification that is used
```
[the remark this answers is not in the document]
Comment (mattfarina) [reply cmd/helm/root.go:61]: Fixed. thanks for catching that.

File: cmd/helm/root.go [hunk 2]
```diff
@@ -46,20 +46,20 @@ Environment variables:
...
+- When no other location is set a default location is used based on operating system
```
[the remark this answers is not in the document]
Comment (mattfarina) [reply cmd/helm/root.go:63]: Updated. Thanks for improving on my grammar.

File: pkg/helmpath/lazypath.go [hunk 3]
```diff
@@ -20,11 +20,34 @@ import (
...
 // lazypath is an lazy-loaded path buffer for the XDG base directory specification.
 type lazypath string
 
-func (l lazypath) path(envVar string, defaultFn func() string, elem ...string) string {
-	base := os.Getenv(envVar)
+func (l lazypath) path(HelmEnvVar, XDGEnvVar string, defaultFn func() string, elem ...string) string {
+
+	// There is an order to checking for a path.
+	// 1. See if a Helm specific environment variable has been set.
+	// 2. Check of an XDG environment variable is set
```
[the remark this answers is not in the document]
Comment (mattfarina) [reply pkg/helmpath/lazypath.go:44]: Fixed. thanks

File: pkg/helmpath/lazypath.go [hunk 4]
```diff
@@ -20,11 +20,34 @@ import (
...
+	// for the data directory. When no value is set a default is used.
+	DataHomeEnvVar = "HELM_DATA_HOME"
+)
+
 // lazypath is an lazy-loaded path buffer for the XDG base directory specification.
 type lazypath string
 
-func (l lazypath) path(envVar string, defaultFn func() string, elem ...string) string {
-	base := os.Getenv(envVar)
+func (l lazypath) path(HelmEnvVar, XDGEnvVar string, defaultFn func() string, elem ...string) string {
```
[the remark this answers is not in the document]
Comment (mattfarina) [reply pkg/helmpath/lazypath.go:40]: It is unusual. I didn't even realize I'd done it. I'm used to my editor yelling at me to fix that. Thanks for catching my oops.

**Comment by mattfarina** [comment]
Updated per feedback from Marc

File: pkg/helmpath/lazypath.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (marckhouzam) [reply pkg/helmpath/lazypath.go]: How about the `XDGEnvVar` parameter?  Is that one ok?

BTW, thanks for not resolving conversations yourself.  I think the person who opened the comment should be the one resolving (maybe that is a well-known practice, but not everyone does it, it seems).

[the remark this answers is not in the document]
Comment (mattfarina) [reply pkg/helmpath/lazypath.go]: XDG is an acronym for X Desktop Group. It should be lowercase and I just did that.

**Comment by marckhouzam** [comment]
LGTM, but (as mentioned before) I'm too new to give approval for this.

File: cmd/helm/root.go [hunk 5]
```diff
@@ -45,19 +45,19 @@ Environment variables:
...
 | $HELM_NO_PLUGINS                   | disable plugins. Set HELM_NO_PLUGINS=1 to disable plugins.                        |
 | $KUBECONFIG                        | set an alternative Kubernetes configuration file (default "~/.kube/config")       |
 
-Helm stores configuration based on the XDG base directory specification, so
+Helm stores cache, configuration, and data based on the following configuration order:
 
-- cached files are stored in $XDG_CACHE_HOME/helm
-- configuration is stored in $XDG_CONFIG_HOME/helm
-- data is stored in $XDG_DATA_HOME/helm
+- If a HELM_*_HOME environment variable is set, it will be used
```
Comment (bacongobbler) [inline cmd/helm/root.go:58]: Should there be periods at the end of these sentences?

**Review by bacongobbler — APPROVED** [review]
code LGTM! This looks like the right approach to me. :)

I had one comment about the grammar being used in the help output, but feel free to ignore if there's precedent for that styling

File: cmd/helm/root.go (hunk 5 above)
Comment (mattfarina) [reply cmd/helm/root.go:58]: I just looked. The other cases of lists do not end with a `.`. To be consistent we should either add them in the other places or keep them off here. I'll keep them off here for the scope of this PR.

**Pull request merged**